docs/cli/plugin_implementation_guide.md

```diff
@@ -39,12 +41,55 @@ filesystem repository in the creation and testing of their feature.
 
 #### Adding Admin Repository
 
+NOTE: This is not applicable if context-aware plugin discovery is enabled within Tanzu CLI.
+
 The admin repository contains the Builder plugin - a plugin which helps scaffold and compile plugins.
 
 To add the admin repository use `tanzu plugin repo add -n admin -b tanzu-cli-admin-plugins -p artifacts-admin`
 
 To add the builder plugin use `tanzu plugin install builder`
 
+#### Plugin Discovery Source
```

Comment:
It seems like this is an H4 under the Repository H3. Is that accurate? I think I'd expect this to be a "sibling" section.